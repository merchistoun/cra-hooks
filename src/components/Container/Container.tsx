import React, { useEffect } from "react";
import { useFibonacci } from "../../hooks";
import { Item } from "../";

export const Container = (): React.ReactElement => {
  const [seq, setSeq] = React.useState<string[]>([]);
  const { fibonacci } = useFibonacci();

  useEffect(() => {
    setSeq(fibonacci(50));
  }, [fibonacci]);

  return (
    <>
      {seq.map((s, i) => (
        <Item value={s} key={i} />
      ))}
    </>
  );
};
