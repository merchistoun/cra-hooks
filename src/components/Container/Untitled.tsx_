import React from "react";

export interface UseHook {}

export const useHook = (): UseHook => {
  // const [state, setState] = React.useState<any>({});
  const value = useRecoilValue(recoilValue);

  // dependencies
  const { function1 } = useHook1();
  const { function2 } = useHook2();
  const { function3 } = useHook3();
  const { function4 } = useHook4();
  const { function5 } = useHook5();
  const { function6 } = useHook6();
  const { function7 } = useHook7();
  const { function8 } = useHook8();

  useEffect(() => {
    if (!value) {
      setValue(doTheWork());
    }
  }, [value]);

  const doTheWork = () => {
    return (
      function1() +
      (function2() / function3()) * function4() -
      function5() +
      (function6() * function7()) / function8()
    );
  };

  return { doTheWork };
};
