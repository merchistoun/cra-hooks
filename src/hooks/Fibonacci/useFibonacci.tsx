import { useSuffix } from "../";

export interface UseFibonacci {
  fibonacci: (n: number) => string[];
}

export const useFibonacci = (): UseFibonacci => {
  const { addSuffix } = useSuffix();

  const fibonacci = (n: number): string[] => {
    const arr = new Array(n);

    for (let i = 0; i < n; i++) {
      arr[i] = i > 1 ? arr[i - 1] + arr[i - 2] : 1;
    }

    return arr.map(addSuffix);
  };
  return { fibonacci };
};
