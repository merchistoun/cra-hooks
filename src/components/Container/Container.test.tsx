import { render } from "@testing-library/react";
import { Container } from "./Container";
import { useFibonacciMock as mockUseFibonacci } from "../../hooks/Fibonacci/useFibonacci.mock";

const mockDependencies = (overrides = {}) => {
  return {
    useFibonacci: mockUseFibonacci(overrides),
    ...overrides,
  };
};

describe("Test component", () => {
  it("should mount without error", async () => {
    const { useFibonacci } = mockDependencies({
      fibonacci: jest
        .fn()
        .mockImplementation(() => ["mock-result-1", "mock-result-2"]),
    });

    const { asFragment } = render(<Container />);
    expect(asFragment()).toMatchSnapshot();

    expect(useFibonacci.fibonacci).toBeCalledTimes(1);
    expect(useFibonacci.fibonacci).toBeCalledWith(50);
  });
});
