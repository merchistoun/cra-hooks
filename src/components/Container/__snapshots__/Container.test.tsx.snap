// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test component should mount without error 1`] = `
<DocumentFragment>
  <div>
    mock-result-1
  </div>
  <div>
    mock-result-2
  </div>
</DocumentFragment>
`;
